platform/linux-generic/odp_ipsec.c

```diff
@@ -1697,7 +1702,7 @@ int odp_ipsec_out_inline(const odp_packet_t pkt_in[], int num_in,
 		}
 
 		if (0 == param->num_opt)
-			opt = NULL;
+			opt = &default_out_opt;
 		else
 			opt = &param->opt[opt_idx];
```

Comment:
`opt = param->num_opt ? &param->opt[opt_idx] : &default_opt_out;` might be simpler here.